tests/read_with_index.rs

```diff
@@ -15,3 +15,12 @@ fn test_line_read_nth() {
 
     assert_eq!(reader.read_nth_shape(1).is_none(), true);
 }
+
+#[test]
```

Comment:
I think this test should be moved as a doctest on the `ShapeReader::shapes_count` method so that its still a test, but it appears on the documentation and illustrate the usage.